spec/models/articles_spec.rb

```diff
@@ -0,0 +1,78 @@
+require 'rails_helper'
+
+RSpec.describe Article, type: :model do
+  describe 'validations' do
+    before { subject.valid? }
+    describe 'title' do
+      context 'when blank' do
+        subject { build(:article, title: nil) }
+        it { expect(subject.errors[:title]).to include("can't be blank") }
+      end
+
+      context 'when less than 5 characters' do
+        subject { build(:article, title: 'name') }
+        it { expect(subject.errors[:title]).to eq(['is too short (minimum is 5 characters)']) }
+      end
+    end
+
+    describe 'category' do
+      context 'when blank' do
+        subject { build(:article, category: nil) }
+        it { expect(subject.errors[:category]).to eq(["can't be blank"]) }
+      end
+    end
+  end
+
+  describe 'scopes' do
+    describe '.ordered' do
+      let!(:older_article) { create(:article, created_at: 23.days.ago) }
+      let!(:newer_article) { create(:article, created_at: 1.days.ago) }
+      subject { described_class.ordered.to_a }
+
+      it 'orders by descending created date' do
+        is_expected.to eq [newer_article, older_article]
+      end
+    end
+  end
+
+  describe '.tagged_with' do
+    context 'when articles with tag exists' do
+      let!(:tag) { create(:tag, :with_articles, name: 'foo') }
+      subject { described_class.tagged_with('foo') }
+
+      it { is_expected.to eq(tag.articles) }
+    end
+
+    context 'when tagged article does not exist' do
+      subject { described_class.tagged_with('foo') }
+      it { expect { subject }.to raise_error(/Couldn't find Tag/) }
+    end
+  end
+
+  describe '#tag_list' do
+    context 'tag list be joined with "," and include all tags names' do
```

Comment:
it is a bad idea to provide here expected result. Context = CONDITION, it = EXPECTED RESULT
So, I expect here:
```ruby
context 'when article with some tags'
context 'when article without tags'

Reply:
done